start-mycroft.sh

```diff
@@ -14,15 +14,16 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-SOURCE="${BASH_SOURCE[0]}"
+# This script is never sourced but always directly executed, so this is safe to do
+SOURCE="$0"
 
 script=${0}
 script=${script##*/}
-cd -P "$( dirname "$SOURCE" )"
+cd -P "$( dirname "$SOURCE" )" || exit
```

Comment:
silent failure? ideally log the error here

Reply:
That's... A bit hard to do. It'll exit with whatever the return code was of `cd` which should be enough indication in itself on what went wrong, but to actually log something is a bit tricky.

https://github.com/koalaman/shellcheck/wiki/SC2242

I'm not entirely sure how to capture the result of `cd` and check it later to then log it in some conditional statement.

Reply:
I do think cd will log the failure (atleast in my tests it does). And as you said exit-status is preserved so this should be fine